orchagent/aclorch.cpp

```diff
@@ -345,11 +346,42 @@ bool AclRule::create()
     {
         SWSS_LOG_ERROR("Failed to create ACL rule");
         AclRange::remove(range_objects, range_object_list.count);
+        decreaseNextHopRefCount();
     }
 
     return (status == SAI_STATUS_SUCCESS);
 }
 
+void AclRule::decreaseNextHopRefCount()
+{
+    if (!m_redirect_target_next_hop.empty())
+    {
+        m_pAclOrch->m_neighOrch->decreaseNextHopRefCount(IpAddress(m_redirect_target_next_hop));
+        m_redirect_target_next_hop.clear();
+    }
+    if (!m_redirect_target_next_hop_group.empty())
+    {
+        IpAddresses target = IpAddresses(m_redirect_target_next_hop_group);
+        m_pAclOrch->m_routeOrch->decreaseNextHopRefCount(target);
+        // remove next hop group in case it's not used by anything else
```

Comment:
I see your point here. A code refactor could be done later to move these part of logic to routeorch. the removal of next hop group could be done after ref count is checked as zero.